From 2d1cd59db1d01ddf313a19b4566d3b730b5ecc20 Mon Sep 17 00:00:00 2001
From: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@intel.com>
Date: Tue, 29 Apr 2014 21:12:15 -0700
Subject: net: rfkill: gpio: Use gpiod cansleep routines when needed

Using regular gpiod set/get routines when the gpio chip "can_sleep"
flag set is not right and it will also cause unncessary warnings.
This patch adds a wrapper thats checks the gpio chip can_sleep
flag and uses gpiod_*_cansleep routines when needed.

Signed-off-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@intel.com>
---
 net/rfkill/rfkill-gpio.c |   16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/net/rfkill/rfkill-gpio.c b/net/rfkill/rfkill-gpio.c
index c4d7c6b..9137658 100644
--- a/net/rfkill/rfkill-gpio.c
+++ b/net/rfkill/rfkill-gpio.c
@@ -41,20 +41,28 @@ struct rfkill_gpio_data {
 	bool			clk_enabled;
 };
 
+static void rfkill_set_gpio(struct gpio_desc *desc, int value)
+{
+	if (gpiod_cansleep(desc))
+		gpiod_set_value_cansleep(desc, value);
+	else
+		gpiod_set_value(desc, value);
+}
+
 static int rfkill_gpio_set_power(void *data, bool blocked)
 {
 	struct rfkill_gpio_data *rfkill = data;
 
 	if (blocked) {
-		gpiod_set_value(rfkill->shutdown_gpio, 0);
-		gpiod_set_value(rfkill->reset_gpio, 0);
+		rfkill_set_gpio(rfkill->shutdown_gpio, 0);
+		rfkill_set_gpio(rfkill->reset_gpio, 0);
 		if (!IS_ERR(rfkill->clk) && rfkill->clk_enabled)
 			clk_disable(rfkill->clk);
 	} else {
 		if (!IS_ERR(rfkill->clk) && !rfkill->clk_enabled)
 			clk_enable(rfkill->clk);
-		gpiod_set_value(rfkill->reset_gpio, 1);
-		gpiod_set_value(rfkill->shutdown_gpio, 1);
+		rfkill_set_gpio(rfkill->reset_gpio, 1);
+		rfkill_set_gpio(rfkill->shutdown_gpio, 1);
 	}
 
 	rfkill->clk_enabled = blocked;
-- 
1.7.9.5

